automation/taskcluster/decision_task.py

```diff
@@ -48,24 +54,41 @@ def pr_or_push(is_master_push):
         print("Exit")
         return {}
 
+    variants = get_build_variants()
+    geckoview_nightly_version = get_geckoview_versions()['nightly']
+    mozharness_task_id = fetch_mozharness_task_id(geckoview_nightly_version)
+    gecko_revision = taskcluster.Queue().task(mozharness_task_id)['payload']['env']['GECKO_HEAD_REV']
+
     build_tasks = {}
     signing_tasks = {}
     other_tasks = {}
 
-    for variant in build_variants.from_gradle():
+    for variant in variants:
         assemble_task_id = taskcluster.slugId()
         build_tasks[assemble_task_id] = BUILDER.craft_assemble_task(variant)
         build_tasks[taskcluster.slugId()] = BUILDER.craft_test_task(variant)
 
-        arch, build_type = get_architecture_and_build_type_from_variant(variant)
+        architecture, build_type = get_architecture_and_build_type_from_variant(variant)
         # autophone only supports arm and aarch64, so only sign/perftest those builds
         if (
+            is_master_push and
             build_type == 'releaseRaptor' and
-            arch in ('arm', 'aarch64') and
-            is_master_push
+            architecture in ('arm', 'aarch64') and
+            SHORT_HEAD_BRANCH == 'master'
```

Comment:
Why do you need both `is_master_push` and `SHORT_HEAD_BRANCH == 'master'`? Aren't they confirming the same underlying fact?

Reply:
Good catch. `is_master_push` is misleading. I just renamed it into `is_push`